tests/conftest.py

```diff
@@ -15,6 +15,32 @@
 
 from . import helpers
 from .helpers.printer import wait_for_output
+import pytest_timeout
+import tempfile
+
+_original_dump_stacks = pytest_timeout.dump_stacks
+
+
+def _print_pydevd_log(reason):
+    sys.stderr.write('\n******************************************************************\n')
+    sys.stderr.write('pydevd log on %s\n' % (reason,))
+    sys.stderr.write('******************************************************************\n')
+    current_pydevd_debug_file = os.environ.get('PYDEVD_DEBUG_FILE')
+    if current_pydevd_debug_file:
+        if os.path.exists(current_pydevd_debug_file):
+            with open(current_pydevd_debug_file, 'r') as stream:
+                sys.stderr.write(stream.read())
+    sys.stderr.write('\n******************************************************************\n')
+    sys.stderr.write('******************************************************************\n')
+
+
+def _on_dump_stack_print_pydevd_log():
+    # On timeout we also want to print the pydev log.
+    _print_pydevd_log('timeout')
+    _original_dump_stacks()
+
+
+pytest_timeout.dump_stacks = _on_dump_stack_print_pydevd_log
```

Comment:
Note that there's an issue open in `pytest-timeout`: https://bitbucket.org/pytest-dev/pytest-timeout/issues/15/new-hook-called-right-before-timeout to support that better (but until that's done, monkeypatching `dump_stacks` seems to be a reasonable workaround).